### PR TITLE
Usage of proprietary headers defined by Fashion Platform GW components

### PR DESCRIPTION
**Pull request by tfrauenstein**
fix: https://github.com/zalando/restful-api-guidelines/issues/554 and https://github.com/zalando/restful-api-guidelines/issues/594

Details and reasoning see [Internal link](https://docs.google.com/document/d/1uuH7GTcz4th6WFXDor386tWbl771tqqccQCESil99EA/edit#heading=h.e1ihg7jrmcij

### CODE REVIEW

File: chapters/proprietary-headers.adoc [hunk 1]
```diff
@@ -101,18 +95,24 @@ disables it in the settings for respective mobile platform.
 *Exception:* The only exception to this guideline are the conventional
 hop-by-hop `X-RateLimit-` headers which can be used as defined in <<153>>.
 
+*Hint:* This guideline does not standardize proprietary headers for 
+our specific gateway components (use case b) above).
```
Comment (tkrop) [inline chapters/proprietary-headers.adoc:99]: `(use case b) above)` ... can we avoid this unbalanced brackets?

File: chapters/proprietary-headers.adoc [hunk 2]
```diff
@@ -10,35 +10,30 @@ methods.
...
+Headers are tyically used to implement protocol processing aspects, such as flow control,
+content negotiation, and authentication, and represent business agnostic 
+request modifiers that provide generic context information ({RFC-7231}#section-5[RFC 7231]).
+
+However, the exceptional usage of proprietary headers is still helpful 
+in situations where domain-specific generic context information 
+a) needs to be passed end to end along the service call chain (even if 
+not all called services use it as input for steering service behavior 
+e.g. `x-sales-channel` header) and/or b) is provided by specific gateway components, 
+for instance, our Fashion Shop API or Merchant API gateway.
```
[the remark this answers is not in the document]
Comment (tfrauenstein) [reply chapters/proprietary-headers.adoc:28]: 👍

File: chapters/proprietary-headers.adoc [hunk 3]
```diff
@@ -10,35 +10,30 @@ methods.
...
+not all called services use it as input for steering service behavior 
+e.g. `x-sales-channel` header) and/or b) is provided by specific gateway components, 
+for instance, our Fashion Shop API or Merchant API gateway. 
+Below, we explicitly define the list of proprietary header exceptions usable for 
+all services for passing through generic context information of our fashion domain (use case a). 
+
+Per convention, non standardized, proprietary header names are prefixed with  `X-`. 
+(Due to backward compatibility, we did not use e.g. `Zalando-` prefix and 
+not followed the Internet Engineering Task Force’s recommendation in {RFC-6648}[RFC 6648] 
+to deprecate usage of  `X-` headers).
```
[the remark this answers is not in the document]
Comment (tfrauenstein) [reply chapters/proprietary-headers.adoc:35]: 👍

File: chapters/proprietary-headers.adoc (hunk 1 above)
Comment (tfrauenstein) [reply chapters/proprietary-headers.adoc:99]: 👍

**Comment by tkrop** [comment]
:+1:

**Comment by ePaul** [comment]
:+1:

**Pull request merged**